rgdps/api/gd/user_relationships.py

```diff
@@ -0,0 +1,282 @@
+from __future__ import annotations
+
+from fastapi import Depends
+from fastapi import Form
+
+from rgdps import logger
+from rgdps.api import responses
+from rgdps.api.context import HTTPContext
+from rgdps.api.dependencies import authenticate_dependency
+from rgdps.common import gd_obj
+from rgdps.common.validators import Base64String
+from rgdps.constants.errors import ServiceError
+from rgdps.constants.users import UserRelationshipType
+from rgdps.models.user import User
+from rgdps.usecases import friend_requests
+from rgdps.usecases import user_relationships
+
+PAGE_SIZE = 10
+
+
+async def friend_requests_get(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    page: int = Form(..., alias="page"),
+    is_sender_user_id: bool = Form(0, alias="getSent"),
+):
+
+    result = await friend_requests.get_user(
+        ctx,
+        user.id,
+        page,
+        PAGE_SIZE,
+        is_sender_user_id=is_sender_user_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to view friend friend requests of {user.id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    if not result.requests:  # No requests found.
+        return responses.code(-2)
+
+    response = "|".join(
+        gd_obj.dumps(
+            [
+                gd_obj.create_friend_request(request.request),
+                gd_obj.create_friend_request_author_string(request.user),
+            ],
+        )
+        for request in result.requests
+    )
+    response += "#" + gd_obj.create_pagination_info(result.total, page, PAGE_SIZE)
+
+    logger.info(f"{user} successfully viewed friend requests.")
+    return response
+
+
+async def friend_request_upload(
```

Comment:
Would call it `friend_request_post` for consistency's sake

Reply:
Resolved in 5ef80ff